### ISSUE TITLE
Update limit on file descriptors in emscripten select syscall

### ISSUE THREAD

**Issue by MarkMcCaskey** [wasmerio/wasmer#482]
from https://spectrum.chat/wasmer/general/nginx-example-crash~da9be763-0e33-4cc6-ad7d-12a668ff15e7

I don't have context here, but from reading https://web.archive.org/web/20160409222737/http://man7.org/linux/man-pages/man2/select.2.html
I have the impression that Emscripten chose 64 for some reason and that what this is compiled with expects like 1024 or something, for nfds.

This is mostly baseless speculation, so be sure to verify this before attempting to fix it.

**Comment by notmaxx** [comment]
thanks @MarkMcCaskey for creating the issue. Not sure if it is related but I am found another crash by doing another load test

```
2019/06/04 14:15:08 [info] 31328#0: *8 client 127.0.0.1 closed keepalive connection
thread 'main' panicked at 'slice index starts at 540684323 but ends at 16777216', src/libcore/slice/mod.rs:2575:5
stack backtrace:
0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
1: std::sys_common::backtrace::_print
2: std::panicking::default_hook::{{closure}}
3: std::panicking::default_hook
4: std::panicking::rust_panic_with_hook
5: std::panicking::continue_panic_fmt
6: rust_begin_unwind
7: core::panicking::panic_fmt
8: core::slice::slice_index_order_fail
9: wasmer_emscripten::syscalls::unix::_syscall102
10: std::panicking::try::do_call
11: rust_maybe_catch_panic
12: <FN as wasmer_runtime_core::typed_func::ExternalFunction<(A, B),Rets>>::to_raw::wrap
"RuntimeError: \"slice index starts at 540684323 but ends at 16777216\""
```
i've got this with `hey -n 100 -c 10 http://localhost:8080/` - in this case `hey` finished successfully but `nginx` crashed

**Comment by MarkMcCaskey** [comment]
@notmaxx Thanks!

I was able to reproduce both errors.  Additionally, by changing things to make them consistent, either by:
- updating rlimit to return 64 (what emscripten hard coded the upper limit to) and nginx's config to only use 64 `worker_connections`
- bumping up the limit to 1024

I get the same error as your second example running the first command.

I think that's probably a good sign that we fixed the first part and that there's an additional bug....
